app/sync.js

```diff
@@ -349,7 +349,8 @@ module.exports.onSyncReady = (isFirstRun, e) => {
       }
     }
     e.sender.send(syncMessages.FETCH_SYNC_RECORDS, categoryNames, startAt)
-    startAt = syncUtil.now()
+    // Reduce syncUtil.now() by this amount to compensate for records pending S3 consistency. See brave/sync #139
+    startAt = syncUtil.now() - config.fetchOffset
```

Comment:
shouldn't you get the new startAt time (in a tmp var) before sending the `FETCH_SYNC_RECORDS` request? It's extremely unlikely now that you've added the offset, but in a resource starved machine there is still a potential for a gap between the sync end/next start times